__src__/kalman_filter/EKF/EKF.cpp

```diff
@@ -0,0 +1,142 @@
+#include "EKF.h"
+#include <cmath>
+
+using namespace Eigen;
+
+EKF::EKF() {
+    Vector4d quaternion(1, 0, 0, 0);
+    Vector3d bias(0, 0, 0);
+
+    xHat.resize(7);
+    xHat << quaternion, bias;
+    yHatBar.resize(3);
+    yHatBar.setZero();
+    p = MatrixXd::Identity(7, 7) * 0.01;
+    Q = MatrixXd::Identity(7, 7) * 0.001;
+    R = MatrixXd::Identity(6, 6) * 0.1;

```

Comment:
### Shortcomings of my previous implementation.
This can be can be memory-intensive! Fixed-size matrices could be used instead of dynamically sized `MatrixXd` to save memory and potentially reduce the footprint.